### PR TITLE
[IMP] open_academy: inheritance t#59945

### PR DESCRIPTION
**Pull request by JJRR5**
**Finished the Inheritance Section of Building a Module Course
The main goal of this sections was to understand how to inherit models and views.**

- Altered existing content using model inheritance, modified the existing partner model and added an instructor boolean field, and a many2many field that corresponds to the session-partner relation
- Displayed these fields on the partner form view
- Added domains on relations fields (instructor)
- Created new fields to partner categories (teacher/level1, teacher/level2). 
- Learned how to use xpath 

**Image 1.- Res Partner View**
image

### CODE REVIEW

**Comment by JJRR5** [comment]
@sebasdrk17 PR upload, could you please check?

File: open_academy/demo/res_partner_demo.xml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<?xml version='1.0' encoding='utf-8'?>
+<odoo noupdate="0">
+    <record id="res_partner_category_teacher_level1" model="res.partner.category">
+        <field name="name">Teacher / Level 1</field>
+    </record>
+    <record id="res_partner_category_teacher_level2" model="res.partner.category">
+        <field name="name">Teacher / Level 2</field>
+    </record>
+</odoo>
+
```
Comment (sebasdrk17) [inline open_academy/demo/res_partner_demo.xml]: remove this file, you don't need it

File: open_academy/__manifest__.py [hunk 2]
```diff
@@ -21,10 +21,12 @@
...
         'views/session_views.xml',
-        'views/course_views.xml'
+        'views/course_views.xml',
+        'views/partner_views.xml'
     ],
     # only loaded in demonstration mode
     'demo': [
-        'demo/demo.xml',
+        'demo/course_demo.xml',
+        'demo/res_partner_demo.xml'
```
Comment (sebasdrk17) [inline open_academy/__manifest__.py]: remove this file

File: open_academy/models/partner.py [hunk 3]
```diff
@@ -0,0 +1,10 @@
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+    instructor = fields.Boolean()
+    teacher_type = fields.Selection(
```
Comment (sebasdrk17) [inline open_academy/models/partner.py]: Leave a blank space to separate the fields
```suggestion
    _inherit = 'res.partner'
    
    instructor = fields.Boolean()
    teacher_type = fields.Selection(
```

File: open_academy/models/partner.py [hunk 4]
```diff
@@ -0,0 +1,10 @@
+from odoo import fields, models
+
+
+class ResPartner(models.Model):
+    _inherit = 'res.partner'
+    instructor = fields.Boolean()
```
Comment (sebasdrk17) [inline open_academy/models/partner.py]:
```suggestion
    is_instructor = fields.Boolean()
```

File: open_academy/models/partner.py [hunk 5]
```diff
@@ -0,0 +1,10 @@
...
+    teacher_type = fields.Selection(
+        selection=[('level1', 'Level 1'), ('level2', 'Level 2')],
+    )
+    session_ids = fields.Many2many('session', string='Relation')
```
Comment (sebasdrk17) [inline open_academy/models/partner.py]: remove the string it is not needed and is not a good name for a relational field

if you read it without context it only says relation, but relation to what?

```suggestion
    session_ids = fields.Many2many('session')
```

File: open_academy/models/session.py [hunk 6]
```diff
@@ -7,6 +7,6 @@ class Session(models.Model):
     start_date = fields.Date()
     duration = fields.Integer()
     number_of_seats = fields.Integer()
-    instructor_id = fields.Many2one('res.partner')
+    instructor_id = fields.Many2one('res.partner', domain='["|",("instructor","=",True),("teacher_type","!=",False)]')
```
Comment (sebasdrk17) [inline open_academy/models/session.py]:
```suggestion
    instructor_id = fields.Many2one('res.partner', domain="['|',('instructor','=',True),('teacher_type','!=',False)]')
```

Comment (sebasdrk17) [reply open_academy/models/session.py]: You need to run pre-commit by the way, the domain is not properly separated

File: open_academy/views/partner_views.xml [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+                                <field name="instructor" string="Is Instructor?"/> 
+                            </group>
+                            <group>
+                                <field name="teacher_type"/>
+                            </group>
+                            <group>
+                                <field name="session_ids"/>
+                            </group>
+                        </page>
+                </xpath>
```
Comment (sebasdrk17) [inline open_academy/views/partner_views.xml]: I have the impression that this is over indented, please run pre-commit

**Review by sebasdrk17 — CHANGES_REQUESTED** [review]
Made comments

File: open_academy/models/partner.py (hunk 3 above)
Comment (JJRR5) [reply open_academy/models/partner.py]: But this a violation to flake8 and pylynt rules, isn't it?

File: open_academy/models/session.py (hunk 6 above)
Comment (JJRR5) [reply open_academy/models/session.py]: the pre-commit didn't mark that error

**Comment by JJRR5** [comment]
@sebasdrk17 changes proved and done, could you please check?

File: open_academy/__manifest__.py [hunk 8]
```diff
@@ -21,10 +21,11 @@
         'security/ir.model.access.csv',
         'views/open_academy_views.xml',
         'views/session_views.xml',
-        'views/course_views.xml'
+        'views/course_views.xml',
+        'views/partner_views.xml'
```
Comment (andreaarce) [inline open_academy/__manifest__.py]:
```suggestion
        'views/res_partner_views.xml'
```

File: open_academy/models/partner.py (hunk 3 above)
Comment (sebasdrk17) [reply open_academy/models/partner.py]: Did the linter mark this as an error? If it did, could you send a screenshot please?

File: open_academy/models/session.py (hunk 6 above)
Comment (sebasdrk17) [reply open_academy/models/session.py]: I think that I make the suggestion wrong, I was saying that the spaces between every coma is missing

it should be 

`instructor_id = fields.Many2one('res.partner', domain="['|', ('instructor', '=', True), ('teacher_type', '!=', False)]')`

File: open_academy/models/partner.py (hunk 3 above)
Comment (JJRR5) [reply open_academy/models/partner.py]: I was wrong, the change is done

**Comment by JJRR5** [comment]
Changes done, @sebasdrk17 could you please check?

File: open_academy/models/__init__.py [hunk 9]
```diff
@@ -1,2 +1,3 @@
 from . import course
 from . import session
+from . import partner
```
Comment (sebasdrk17) [inline open_academy/models/__init__.py:3]: res_partner

Comment (sebasdrk17) [inline open_academy/models/__init__.py]:
```suggestion
from . import res_partner
```

File: open_academy/models/partner.py [hunk 10]
```diff
@@ -0,0 +1,11 @@
+from odoo import fields, models
+
```
Comment (sebasdrk17) [inline open_academy/models/partner.py]: change the name of this file to be res_partner.py please

**Comment by JJRR5** [comment]
@sebasdrk17 could you please check?

**Review by sebasdrk17 — APPROVED** [review]
@andreaarce LGTM, could you check please?

File: open_academy/models/session.py [hunk 11]
```diff
@@ -7,6 +7,6 @@ class Session(models.Model):
     start_date = fields.Date()
     duration = fields.Integer()
     number_of_seats = fields.Integer()
-    instructor_id = fields.Many2one('res.partner')
+    instructor_id = fields.Many2one('res.partner', domain='["|", ("is_instructor","=",True), ("teacher_type","!=",False)]')
     course_id = fields.Many2one('course', required=True)
     attendees_ids = fields.Many2many('res.partner', string="Attendees")
```
Comment (andreaarce) [inline open_academy/models/session.py]: Remove the string, and test if the label on the form still appears as Attendees AFAIK the default string will be attendees and the string is not necessary please test it.

Comment (JJRR5) [reply open_academy/models/session.py]: done

**Review by andreaarce — APPROVED** [review]
@sebasdrk17 LGTM

**Comment by sebasdrk17** [comment]
@JJRR5 please squash your commits and rebase to 15.0 if needed

**Comment by JJRR5** [comment]
Squash done @sebasdrk17

**Pull request merged**